lib/improver/argparser.py

```diff
@@ -28,19 +28,185 @@
 # CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
-"""Common option utilities for Improver CLIs."""
+"""Common option utilities for improver CLIs."""
 
-import argparse
+from argparse import ArgumentParser
 
 
-class ArgParser(argparse.ArgumentParser):
-    """Argument parser for improver CLIs"""
+class ArgParser(ArgumentParser):
+    """Argument parser for improver CLIs.
+    (Subclass of argparse.ArgumentParser)"""
+
+    # CENTRALIZED_ARGUMENTS, is a dictionary where, for each element:
+    #   the key is a string representing the argument name,
+    #   the value is a list containing 2 elements:
+    #       1. a list of strings containing the different flags
```

Comment:
I'm not quite clear on what "flags" are.  Could we have an example of the usage in this docstring (eg as at line 61).

Reply:
Good point - will do. They're named as "names or flags" in the argparse.ArgumentParser documentation, and given that I was using "names" to refer to the keys of the dictionary, I thought "flags" would be a better descriptor, but realised it wasn't ideal.